### PR TITLE
Update build scripts and add debug symbols to fix the build archive step

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by cdhoffmann — COMMENTED** [review]
Thank you for working on this. The only comment i have is about the changes in the readme. I think it could be worded a little better. Something like:
"First run `make pod-install` to ensure dependencies are installed. Then run `make archive` from the root directory to generate `.xcframeworks` for each module under the `build` folder. Drag and drop all `.xcframeworks` to your app target in Xcode.

**Comment by emdobrin** [comment]
@cdhoffmann the archive command also runs pod update https://github.com/adobe/aepsdk-analytics-ios/pull/57/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R33 so I think this additional comment is not necessary

**Comment by cdhoffmann** [comment]
@emdobrin Good call. We can probably just leave the readme as it was before this PR then.

**Review by cdhoffmann — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -29,28 +30,25 @@ test:
...
-	xcodebuild -create-xcframework -framework $(SIMULATOR_ARCHIVE_PATH)$(PROJECT_NAME).framework -framework $(IOS_ARCHIVE_PATH)$(PROJECT_NAME).framework -output ./build/$(PROJECT_NAME).xcframework
+	xcodebuild archive -workspace $(PROJECT_NAME).xcworkspace -scheme $(SCHEME_NAME_XCFRAMEWORK) -archivePath "./build/ios_simulator.xcarchive" -sdk iphonesimulator -destination="iOS Simulator" SKIP_INSTALL=NO BUILD_LIBRARIES_FOR_DISTRIBUTION=YES
+	xcodebuild -create-xcframework -framework $(SIMULATOR_ARCHIVE_PATH)$(PROJECT_NAME).framework -debug-symbols $(SIMULATOR_ARCHIVE_DSYM_PATH)$(EXTENSION_NAME).framework.dSYM -framework $(IOS_ARCHIVE_PATH)$(PROJECT_NAME).framework -debug-symbols $(IOS_ARCHIVE_DSYM_PATH)$(EXTENSION_NAME).framework.dSYM -output ./build/$(PROJECT_NAME).xcframework
 
 clean:
 	rm -rf ./build
 
 format:
-	swiftformat . --swiftversion 5.2
+	swiftformat . --swiftversion 5.1
```
Comment (emdobrin) [inline Makefile:42]: do we need to specify the version L42 and L52?

File: Makefile [hunk 2]
```diff
@@ -29,28 +30,25 @@ test:
...
 
 lint-autocorrect:
-	swiftlint autocorrect
+	./Pods/SwiftLint/swiftlint autocorrect
```
Comment (emdobrin) [inline Makefile:45]: does the xcode project have swiftlint setup too or does it run on make command only in this repo?

File: Makefile (hunk 1 above)
Comment (addb) [reply Makefile:42]: Yes we need the version 
`warning: No Swift version was specified, so some formatting features were disabled. Specify the version of Swift you are using with the --swiftversion option, or by adding a .swift-version file to your project.`

**Review by emdobrin — APPROVED** [review]

**Pull request merged**